test/hexpm/web/controllers/api/release_controller_test.exs

```diff
@@ -524,4 +524,95 @@ defmodule Hexpm.Web.API.ReleaseControllerTest do
       assert result["version"] == "0.0.1"
     end
   end
+
+  describe "GET /api/packages/:name/releases/:version/downloads" do
+    setup do
+      user = insert(:user)
+      repository = insert(:repository)
+      package = insert(:package, package_owners: [build(:package_owner, owner: user)])
+      release = insert(:release, package: package, version: "0.0.1")
+
+      insert(:download, release: release, downloads: 1, day: ~D[2000-01-01])
+      insert(:download, release: release, downloads: 1, day: ~D[2000-02-01])
+      insert(:download, release: release, downloads: 1, day: ~D[2000-02-07])
+      insert(:download, release: release, downloads: 1, day: ~D[2000-02-08])
+
+      %{
+        user: user,
+        repository: repository,
+        package: package,
+        release: release
+      }
+    end
+
+    test "get release downloads (all by default)", %{package: package, release: release} do
+      result =
+        build_conn()
+        |> get("api/packages/#{package.name}/releases/#{release.version}/downloads")
+        |> json_response(200)
+
+      assert result["name"] == "#{package.name}"
+      assert result["version"] == "#{release.version}"
+      assert result["downloads"] == 4
+
+      result =
+        build_conn()
+        |> get("api/packages/#{package.name}/releases/#{release.version}/downloads/all")
+        |> json_response(200)
+
+      assert result["name"] == "#{package.name}"
+      assert result["version"] == "#{release.version}"
+      assert result["downloads"] == 4
+
+      result =
+        build_conn()
+        |> get("api/packages/#{package.name}/releases/#{release.version}/downloads/xxx")
+        |> json_response(200)
+
+      assert result["name"] == "#{package.name}"
+      assert result["version"] == "#{release.version}"
+      assert result["downloads"] == 4
+    end
+
+    test "get release downloads by day", %{package: package, release: release} do
+      result =
+        build_conn()
+        |> get("api/packages/#{package.name}/releases/#{release.version}/downloads/by_day")
+        |> json_response(200)
+
+      assert result["name"] == "#{package.name}"
+      assert result["version"] == "#{release.version}"
+      assert result["downloads"] == [
+        ["2000-01-01", 1],
+        ["2000-02-01", 1],
+        ["2000-02-07", 1],
+        ["2000-02-08", 1],
+      ]
+    end
+
+    test "get release downloads by week", %{package: package, release: release} do
+      result =
+        build_conn()
+        |> get("api/packages/#{package.name}/releases/#{release.version}/downloads/by_week")
+        |> json_response(200)
+
+      assert result["name"] == "#{package.name}"
+      assert result["version"] == "#{release.version}"
+      assert length(result["downloads"]) == 3 # to avoid different LOCALE (week)
```

Comment:
Different locale shouldn't matter. We should expect everyone to have the same locale in the database or other things will break anyway.

Reply:
Ok, but my locale's starting day of the week is monday, and I know for others is sunday. If I write in the test:
```
      assert result["downloads"] == [
        ["1999-12-27", 1],
        ["2000-01-31", 1],
        ["2000-02-07", 2],
      ]
```
Wouldn't fail for other locales that the monday is not the first day of the week? How would you test it? Where can I find the setting locale of the database (i was checking config/{dev,prod,config}.exs)?

Reply:
The locale would be a setting on the database itself, not something that would be configured on the Elixir project. But I just checked the postgres source code and they follow the ISO standard by starting the week at Monday.